public/app/plugins/panel/graph/module.ts

```diff
@@ -343,6 +343,10 @@ class GraphCtrl extends MetricsPanelCtrl {
   formatDate = (date: DateTimeInput, format?: string) => {
     return this.dashboard.formatDate.apply(this.dashboard, [date, format]);
   };
+
+  getDataFrameByName = (seriesName: string) => {
+    return this.dataList.filter(dataFrame => dataFrame.name === seriesName)[0];
+  };
```

Comment:
I don't totally understand how and where this is called -- but we should be careful not to assume the DataFrame name is unique.  name is an optional string.  refId may be better, but there will often be multiple DataFrames with the same refId.   maybe consider index?

Reply:
This is used in graph vis to get the data frame corresponding to the series being clicked. Wondering, are multiple data frames with same refId valid in graph context?

Reply:
> Wondering, are multiple data frames with same refId valid in graph context?

For sure, in influx the query `SELECT a,b FROM measurement` is returned as two TimeSeries with the same refId.

Reply:
Thanks for clarification